src/imp/unix.rs

```diff
@@ -28,15 +38,23 @@ pub fn cstr(path: &Path) -> io::Result<CString> {
         .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "path contained a null"))
 }
 
+#[cfg(not(target_os = "redox"))]
 pub fn create_named(path: &Path) -> io::Result<File> {
     unsafe {
         let path = try!(cstr(path));
-        match open(path.as_ptr() as *const c_char,
-                   O_CLOEXEC | O_EXCL | O_RDWR | O_CREAT,
-                   0o600) {
-            -1 => Err(io::Error::last_os_error()),
-            fd => Ok(FromRawFd::from_raw_fd(fd)),
-        }
+        let fd = try!(cvt_err(open(path.as_ptr() as *const c_char,
+                                   O_CLOEXEC | O_EXCL | O_RDWR | O_CREAT,
+                                   0o600)));
+        Ok(FromRawFd::from_raw_fd(fd))
+    }
+}
+
+#[cfg(target_os = "redox")]
+pub fn create_named(path: &Path) -> io::Result<File> {
+    unsafe {
+        let fd = try!(cvt_err(open(path.as_os_str().as_bytes(),
+                                   O_CLOEXEC | O_EXCL | O_RDWR | O_CREAT | 0o600)));
```

Comment:
I'm pretty sure the `0o600` is not supposed to be ORed with this. Is there any way to atomically set the permissions on this file?